File: backend/src/main/kotlin/com/backend/models/entities/Match.kt
package com.backend.models.entities

import jakarta.persistence.*
import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import java.time.Instant
import java.util.*

@Entity
@Table(name = "matches")
class Match(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    var id: UUID? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "game_id", nullable = false)
    var game: Game,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id", nullable = false)
    var createdBy: User,

    @Column(name = "is_team_based", nullable = false)
    var isTeamBased: Boolean = false,

    @Column(name = "played_at", nullable = false)
    var playedAt: Instant,

    @Column(name = "duration_minutes", nullable = true)
    var durationMinutes: Int? = null,

    @Column(name = "place", nullable = true)
    var place: String? = null,

    @Column(name = "notes", nullable = true, columnDefinition = "TEXT")
    var notes: String? = null,

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(
        name = "match_expansions",
        joinColumns = [JoinColumn(name = "match_id")],
        inverseJoinColumns = [JoinColumn(name = "expansion_game_id")],
    )
    var expansionsUsed: MutableSet<Game> = mutableSetOf(),

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    val createdAt: Instant = Instant.now(),

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    var updatedAt: Instant = Instant.now(),
)

File: frontend-admin/src/types/match.ts
import type { User } from '@/types/user';
import type { Game } from '@/types/game';

export interface MatchPlayerRef {
    id: string;
    user: User | null;
    guestName: string | null;
}

export interface MatchTeam {
    id: string;
    name: string | null;
    color: string;
    score: number;
    isWinner: boolean;
    startingPosition: number | null;
    players: MatchPlayerRef[];
}

export interface MatchPlayer {
    id: string;
    user: User | null;
    guestName: string | null;
    color: string | null;
    score: number | null;
    isWinner: boolean | null;
    startingPosition: number | null;
}

export interface Match {
    id: string;
    game: Game;
    createdBy: User;
    isTeamBased: boolean;
    playedAt: string;
    place: string | null;
    notes: string | null;
    durationMinutes: number | null;
    createdAt: string;
    teams: MatchTeam[] | null;
    players: MatchPlayer[] | null;
    expansionsUsed: Game[];
}